modules/refined/src/test/scala/ru/tinkoff/phobos/refined/RefinedDecodersSpec.scala

```diff
@@ -0,0 +1,123 @@
+package ru.tinkoff.phobos.refined
+
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.refineMV
+import eu.timepit.refined.string.MatchesRegex
+import eu.timepit.refined.types.numeric.NonNegLong
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import ru.tinkoff.phobos.annotations.{ElementCodec, XmlCodec}
+import ru.tinkoff.phobos.decoding.XmlDecoder
+import ru.tinkoff.phobos.syntax.{attr, text}
+import ru.tinkoff.phobos.testString._
+import shapeless.{Witness => W}
+
+class RefinedDecodersSpec extends AnyWordSpec with Matchers {
+  type NumericAtLeastTo = MatchesRegex[W.`"[0-9]{2,}"`.T]
+
+  @XmlCodec("test")
+  case class Test(x: Int, y: Refined[String, NumericAtLeastTo])
+
+  @ElementCodec
+  case class Foo(@attr bar: Int, @text baz: NonNegLong)
+  @XmlCodec("qux")
+  case class Qux(str: String, foo: Foo)
+
+  "refined decoder" should {
+    "decode element correctly" in {
+
+      val sampleXml = """
+         | <?xml version='1.0' encoding='UTF-8'?>
+         | <test>
+         |   <x>2</x>
+         |   <y>123</y>
+         | </test>
+          """.stripMargin.minimized
+
+      val expectedResult = Test(2, refineMV[NumericAtLeastTo]("123"))
+
+      XmlDecoder[Test].decode(sampleXml) shouldEqual (Right(expectedResult))
+
+    }
+
+    "decode text correctly" in {
+      val sampleXml =
+        """
+          | <?xml version='1.0' encoding='UTF-8'?>
+          | <qux>
+          |   <str>42</str>
+          |   <foo bar="42">1000</foo>
+          | </qux>
+          """.stripMargin.minimized
+
+      val expectedResult = Qux("42", Foo(42, NonNegLong(1000L)))
+      XmlDecoder[Qux].decode(sampleXml) shouldEqual Right(expectedResult)
+    }
+
+    "provide errors with minimal verbosity by default" in {
+      val sampleXml0 = """
+        | <?xml version='1.0' encoding='UTF-8'?>
+        | <test>
+        |   <x>2</x>
+        |   <y>1</y>
+        | </test>
+          """.stripMargin.minimized
+
+      XmlDecoder[Test].decode(sampleXml0).left.get.text shouldEqual """Predicate failed: "1".matches("[0-9]{2,}")."""
```

Comment:
На left.get ругается, как на deptecated, думаю, стоит заменить на
```suggestion
      import cats.syntax.either._
      XmlDecoder[Test].decode(sampleXml0).leftMap(_.text) shouldEqual Left("""Predicate failed: "1".matches("[0-9]{2,}").""")
```